### PR TITLE
Cleanup OvrIntegration issues

### PR DESCRIPTION
**Pull request by Squareys**
Hi @mosra !

I finally got around to cleaning up #13... sorry it took me so long. And sorry for the mess, some of the things I fixed were rather sloppy.

Regards, Jonathan.

PS: Take your time with the review, I will need to compile check the example still anyway, and this shouldn't be urgent in any way.

### CODE REVIEW

File: src/Magnum/OvrIntegration/Compositor.cpp [hunk 1]
```diff
@@ -108,8 +103,6 @@ LayerQuad& LayerQuad::setQuadSize(const Vector2& size) {
     return *this;
 }
 
-Compositor::Compositor() = default;
```
Comment (mosra) [inline src/Magnum/OvrIntegration/Compositor.cpp]: I would keep the constructor here, the class has heavy members (`std::vector`) and having the constructor inlined will increase compile time and binary size.

File: src/Magnum/OvrIntegration/Context.cpp [hunk 2]
```diff
@@ -80,4 +80,15 @@ std::unique_ptr<Session> Context::createSession() {
     return std::unique_ptr<Session>(new Session(session));
 }
 
+Error Context::error() const {
+    ovrErrorInfo info;
+    ovr_GetLastErrorInfo(&info);
+
+    Error err;
+    err.type = ErrorType(info.Result);
+    strncpy(err.message, info.ErrorString, 512);
```
Comment (mosra) [inline src/Magnum/OvrIntegration/Context.cpp]: `std::strncpy()`?

**Comment by mosra** [comment]
Thanks a lot! :+1: 

I still saw mentions of the OVR API instead of the wrapped API in the error enum, can you look there also?

**Comment by Squareys** [comment]
@mosra I addressed the new issues (de-inlined the Compositor constructor, referenced OvrIntegration types in errors enum and added the `std::`). This could possibly be good to go now :)

Regards, Squareys.

**Comment by Squareys** [comment]
@mosra Rebased and removed error enum values to know be OculusSDK 1.6.0 compliant!

I would be really happy, if this could get merged in the next 7 days, if that is possible somehow :)

File: src/Magnum/OvrIntegration/Session.cpp [hunk 3]
```diff
@@ -37,6 +37,29 @@
...
+enum class HmdStatusFlag: UnsignedByte {
+    /**
+     * A mirror texture was created for the HMD and needs to be destroyed on
+     * destruction of the HMD
+     */
+    HasMirrorTexture = 1 << 0,
+
+    /** The HMD was created as a debug HMD (without real hardware) */
+    Debug = ovrHmdCap_DebugDevice /* 0x0010 */
+};
```
Comment (mosra) [inline src/Magnum/OvrIntegration/Session.cpp]: :+1: for the idea of having the enum definition only in the `*.cpp` file. A new thing for me! :)

**Comment by mosra** [comment]
Merged. Thanks a lot!

**Pull request merged**